pkg/sdk/api/v1beta1/common_types.go

```diff
@@ -37,6 +39,41 @@ type Metrics struct {
 
 type KubernetesStorage struct {
 	HostPath *corev1.HostPathVolumeSource `json:"host_path,omitempty"`
+	EmptyDir *corev1.EmptyDirVolumeSource `json:"emptyDir,omitempty"`
+}
+
+// GetVolume returns a default emptydir volume if none configured
+// `logging` is the name of the logging CR to make sure multiple
+//           logging system can coexist in case a hostPath mount is used
+// `name`    will be the name of the volume and the lowest level directory in case a hostPath mount is used
+func (storage *KubernetesStorage) GetVolume(logging, name string) corev1.Volume {
+	const (
+		hostPath = "/opt/logging-operator/%s/%s"
+	)
+	volume := corev1.Volume{
+		Name: name,
+	}
+	if storage != nil {
```

Comment:
We should do this check before invoking the function.

Reply:
modified to be a value receiver